x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts

```diff
@@ -246,6 +246,7 @@ export const createSecurityRuleTypeWrapper: CreateSecurityRuleTypeWrapper =
               await ruleExecutionLogger.logStatusChange({
                 newStatus: RuleExecutionStatusEnum.failed,
                 message: errorMessage,
+                userError: true,
```

Comment:
can be there other cases for this request to fail that can be considered not user error?

we know, if it not found - it's user error.
what if we get some 500 error from ES?

Reply:
Should be fixed here: f9ba64fd0504e07a412f3c3ac0035546ac4f0b04

Reply:
We probably should not rely on `DataViewError` in this case:

```ts
export class DataViewError extends Error {}
```

https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/utils/get_input_output_index.ts#L37

Is this possible to check status_code? If it's 404, we can safely put is user error.

Reply:
Yes definitely! I'll make the changes.

Reply:
Okay I think this should be safer 7a0c3c29bcd288b776dfe36713bdbd41e61d8ff3